### PR TITLE
feat: research output list/detail view toggle

### PR DESCRIPTION
**Pull request by kmaid**

### CODE REVIEW

File: apps/frontend/src/hooks/pagination.ts [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
   const searchParams = new URLSearchParams(useLocation().search);
   const currentPage = Number(searchParams.get('currentPage')) ?? 0;
 
+  const resetPaginationSearchParams = new URLSearchParams(searchParams);
+  resetPaginationSearchParams.delete('currentPage');
+
+  const listView = searchParams.get(viewParam) === listViewValue;
+  const listViewParams = new URLSearchParams(resetPaginationSearchParams);
+  listViewParams.set(viewParam, listViewValue);
+  const cardViewParams = new URLSearchParams(resetPaginationSearchParams);
```
Comment (jeysal) [inline apps/frontend/src/hooks/pagination.ts:19]: detail not card for wording consistency

File: apps/frontend/src/hooks/pagination.ts [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
-export const usePaginationParams = (pageSize = DEFAULT_PAGE_SIZE) => {
+export const usePaginationParams = () => {
   const history = useHistory();
   const searchParams = new URLSearchParams(useLocation().search);
   const currentPage = Number(searchParams.get('currentPage')) ?? 0;
 
+  const resetPaginationSearchParams = new URLSearchParams(searchParams);
+  resetPaginationSearchParams.delete('currentPage');
+
+  const listView = searchParams.get(viewParam) === listViewValue;
```
Comment (jeysal) [inline apps/frontend/src/hooks/pagination.ts:16]: `isListView`

File: packages/react-components/src/icons/card-view.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/* istanbul ignore file */
+
+const cardViewIcon = (
+  <svg
+    width={24}
+    height={24}
+    viewBox="0 0 24 24"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+    stroke="#4D646B"
```
Comment (jeysal) [inline packages/react-components/src/icons/card-view.tsx:12]: shiet just realized we probably have a lot of color duplication in these icons. meh, can probably do one pass replacing them with the defined colors at some point

File: packages/react-components/src/molecules/ListControls.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  listViewHref,
+  cardViewHref,
+}) => (
+  <div css={containerStyles}>
+    <NavHashLink
+      to={cardViewHref}
+      activeClassName={activeClassName}
+      css={[styles, { [`&.${activeClassName}`]: activeStyles }]}
+      smooth
+      isActive={() => !listView}
```
Comment (jeysal) [inline packages/react-components/src/molecules/ListControls.tsx:63]: why is this necessary? The nav link should do that by default based on `to`

File: apps/frontend/src/hooks/pagination.ts (hunk 1 above)
Comment (kmaid) [reply apps/frontend/src/hooks/pagination.ts:19]: I think i snuck in another commit. They changed the designs on me

File: packages/react-components/src/molecules/ListControls.tsx (hunk 4 above)
Comment (jeysal) [reply packages/react-components/src/molecules/ListControls.tsx:63]: that would also reduce props to two

File: packages/react-components/src/molecules/__tests__/ListControls.test.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { StaticRouter } from 'react-router-dom';
+
+import ListControls from '../ListControls';
```
Comment (jeysal) [inline packages/react-components/src/molecules/__tests__/ListControls.test.tsx:5]: test active style (have many examples for that in the codebase)

File: packages/react-components/src/molecules/ListControls.tsx (hunk 4 above)
Comment (kmaid) [reply packages/react-components/src/molecules/ListControls.tsx:63]: The URL will change with pagination and searchQuery params. I don't know how it would match without this

Comment (jeysal) [reply packages/react-components/src/molecules/ListControls.tsx:63]: right, and search query is preserved, but current page isn't I assume?

File: apps/frontend/src/hooks/pagination.ts (hunk 1 above)
Comment (jeysal) [reply apps/frontend/src/hooks/pagination.ts:19]: Well, or if it's card view now, use that term everywhere :D

File: packages/react-components/src/molecules/ListControls.tsx (hunk 4 above)
Comment (jeysal) [reply packages/react-components/src/molecules/ListControls.tsx:63]: The thing is, then we don't actually want a `NavLink`, but just a normal `Link`

Comment (jeysal) [reply packages/react-components/src/molecules/ListControls.tsx:63]: What I would suggest instead though: Still use `NavLink`, and still omit the `listView` prop, but in `isActive` look at that one specific `viewParam` and compare it to determine the active state

Comment (jeysal) [reply packages/react-components/src/molecules/ListControls.tsx:63]: That should mean little change from now because we're still using `NavLink`, but in a way that makes sense (you can tell because it actually uses the param to the isActive callback :D), and we still manage to omit one prop

File: apps/frontend/src/hooks/pagination.ts (hunk 1 above)
Comment (kmaid) [reply apps/frontend/src/hooks/pagination.ts:19]: So the default view is Card View. The new view is List View. I think the naming is consistent

Comment (jeysal) [reply apps/frontend/src/hooks/pagination.ts:19]: But there are still things named `detail` left

File: apps/frontend/src/hooks/__tests__/pagination.test.ts [hunk 6]
```diff
@@ -51,6 +61,33 @@ describe('usePaginationParams', () => {
       urlSearchParamsToObject(result.current.useLocationResult.search),
     ).toEqual({});
   });
+
+  it('generates details and list view params', () => {
```
Comment (jeysal) [inline apps/frontend/src/hooks/__tests__/pagination.test.ts:65]:
```suggestion
  it('generates card and list view params', () => {
```

File: packages/react-components/src/molecules/ListControls.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+const containerStyles = css({
+  display: 'grid',
+  gridAutoFlow: 'column',
+  columnGap: '24px',
+  width: 'fit-content',
+  margin: `${12 / perRem}em 0`,
+});
+
+const styles = css({
```
Comment (jeysal) [inline packages/react-components/src/molecules/ListControls.tsx:17]:
```suggestion
const linkStyles = css({
```

File: packages/react-components/src/molecules/ListControls.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  readonly listViewHref: string;
+  readonly cardViewHref: string;
+}
+const ListControls: React.FC<ListControlsProps> = ({
+  isListView,
+  listViewHref,
+  cardViewHref,
+}) => (
+  <div css={containerStyles}>
+    <Link to={cardViewHref} css={styles}>
```
Comment (jeysal) [inline packages/react-components/src/molecules/ListControls.tsx:56]:
```suggestion
    <Link to={cardViewHref} css={linkStyles}>
```

File: packages/react-components/src/molecules/ListControls.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+      <p css={[textStyles, !isListView && activeStyles]}>
+        <span css={iconStyles}>{cardViewIcon}</span>
+        Card View
+      </p>
+    </Link>
+    <Link to={listViewHref} css={styles}>
```
Comment (jeysal) [inline packages/react-components/src/molecules/ListControls.tsx:62]:
```suggestion
    <Link to={listViewHref} css={linkStyles}>
```

File: packages/react-components/src/molecules/__tests__/ListControls.test.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  const { getByText, rerender } = render(
+    <ListControls
+      cardViewHref="/card?123"
+      listViewHref="/list?321"
+      isListView={false}
+    />,
+    { wrapper: MemoryRouter },
+  );
+  expect(
+    getComputedStyle(getByText(/card/i, { selector: 'p' })).fontWeight,
```
Comment (jeysal) [inline packages/react-components/src/molecules/__tests__/ListControls.test.tsx:30]: Avoid `selector` since it's overly specific (we don't need to test that it's in a `p`), and use `findParentWithStyle`

**Review by jeysal — COMMENTED** [review]
cool, a couple minor local suggestions then lgtm :upside_down_face:

File: packages/react-components/src/molecules/__tests__/ListControls.test.tsx (hunk 10 above)
Comment (kmaid) [reply packages/react-components/src/molecules/__tests__/ListControls.test.tsx:30]: The problem was get by text was picking up the SVG title. Thats why i put the selector

Comment (jeysal) [reply packages/react-components/src/molecules/__tests__/ListControls.test.tsx:30]: Ah ok

**Review by jeysal — APPROVED** [review]
